src/qilib/data_set/data_set.py

```diff
@@ -116,7 +116,11 @@ def add_data(self, index_or_slice: Union[Tuple[int, ...], int], data: Dict[str,
         """
 
         for array_name, data_value in data.items():
-            self._data_arrays[array_name][index_or_slice] = data_value
+            if array_name in self._data_arrays:
+                self._data_arrays[array_name][index_or_slice] = data_value
+            else:
+                self._set_arrays[array_name][index_or_slice] = data_value
```

Comment:
I know this is how it was, that an array name that does not exist raises a KeyError, but since we are now checking if the array is in `self._data_arrays` should we also check if it is in `_set_arrays` and if in neither, raise a more meaningful error? Just a thought.

Reply:
Agree